### PR TITLE
chore: make contour file available offline

### PR DESCRIPTION
**Pull request by odockal**
### What does this PR do?
Makes contour.yaml file generated on demand and permanent part of the repo, available under `extensions/kind/src/resources`. Can be updated by npm task `install:contour`.
### What issues does this PR fix or reference?
#2248 

### How to test this PR?
Kind extension build passes: `yarn build:extensions:kind`

Verify downloading/generating of the file:
Removing the contour available locally: `rm extensions/kind/src/resources`
`yarn build:extensions:kind` -> fails
To download/generate contour:
`yarn --cwd extensions/kind run install:contour`
will be available on `extensions/kind/src/resources/contour.yaml`
Build now passes again: `yarn build:extensions:kind`

Unit tests should pass as well:
`yarn test:extensions:kind`

### CODE REVIEW

**Comment by benoitf** [comment]
Might be easier to just have the file there, not download it each time. And the script to fetch it/update it (like dependabot/renovate to update the file when there is a new contour version)

b/c here I don't see the benefit of downloading it ?

@jeffmaury what is your feedback ?

**Review by jeffmaury — CHANGES_REQUESTED** [review]
Seems to me that your changed the location of the contour.yaml file but the code has not been updated

**Comment by odockal** [comment]
@jeffmaury The local yaml file is decoded and saved to the same location as when downloading. So the location does not change at that end, I also do not have to change vite config module '@gen' this way.

**Comment by jeffmaury** [comment]
After discussion with @benoitf this PR should be rewritten so that:

- contour file is now part of the source tree and code should be updated so use that file (moved from src-generated to src)
- download script should be updated so that it is now used to update the file in the source tree
- download script should not be called automatically
- src-generated should be removed

**Comment by odockal** [comment]
@jeffmaury @benoitf I have updated the PR and also a PR description -> should have updated steps to verify. I would also like to test the actual functionality of the contour ingress node by building app locally and testing it with a deployment of some app but my laptop is somehow messed today - so I only get too many open files on the log of the app deployed to the kind cluster and cannot check app is available on the `localhost:9090`.

**Review by jeffmaury — CHANGES_REQUESTED** [review]
.gitignore should be deleted

**Review by jeffmaury — APPROVED** [review]
LGTM

**Comment by benoitf** [comment]
FYI @dgolovin @cdrage I don't know who will do the release but with that PR being merged

https://github.com/flathub/io.podman_desktop.PodmanDesktop/blob/master/io.podman_desktop.PodmanDesktop.yml#L65-L66

and https://github.com/flathub/io.podman_desktop.PodmanDesktop/tree/master/patches folder will need to be removed when updating flatpak repository for cutting the flatpak release.

File: extensions/kind/scripts/download.ts [hunk 1]
```diff
@@ -68,10 +67,12 @@ async function download(tagVersion: string, repoPath: string, fileName: string):
...
 
   fs.writeFileSync(destFile, buffer);
-  console.log(`Downloaded to ${destFile}`);
+  console.log(`Contour.yaml available at ${destFile}`);
 }
 
 // grab the manifests from the given URL
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 // download the file from the given URL and store the content in destFile
+// partiullar contour file should be manually added to the repo once downloaded
```
Comment (benoitf) [inline extensions/kind/scripts/download.ts]:
```suggestion
// particular contour file should be manually added to the repo once downloaded
```

**Review by benoitf — APPROVED** [review]
there is a typo but it does not prevent the approval

**Comment by odockal** [comment]
@benoitf @jeffmaury thanks for review, I have enabled automerge and fixed typo by suggestion. thx

**Pull request merged**